show-geodesic-path-between-two-points/src/main/java/com/esri/arcgismaps/sample/showgeodesicpathbetweentwopoints/MainActivity.kt

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2023 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.showgeodesicpathbetweentwopoints
+
+import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.lifecycleScope
+import com.arcgismaps.ApiKey
+import com.arcgismaps.ArcGISEnvironment
+import com.arcgismaps.Color
+import com.arcgismaps.geometry.LinearUnit
+import com.arcgismaps.geometry.LinearUnitId
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.geometry.GeometryEngine
+import com.arcgismaps.geometry.Polyline
+import com.arcgismaps.geometry.GeodeticCurveType
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbol
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbolStyle
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.esri.arcgismaps.sample.showgeodesicpathbetweentwopoints.databinding.ActivityMainBinding
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.launch
+import kotlin.math.roundToInt
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    // set up data binding for the activity
+    private val activityMainBinding: ActivityMainBinding by lazy {
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    // set up data binding for the mapView
+    private val mapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    // shows the distance information as a textview
+    private val infoTextView by lazy {
+        activityMainBinding.infoTextView
+    }
+
+    // a red marker symbol for the location points
+    private val locationMarkerSymbol by lazy {
+        SimpleMarkerSymbol(
+            SimpleMarkerSymbolStyle.Circle,
+            Color.red,
+            10f
+        )
+    }
+
+    // the marker graphic for the starting location
+    private val startingLocationMarkerGraphic by lazy {
+        Graphic().apply {
+            symbol = locationMarkerSymbol
+            geometry = startingPoint
+        }
```

Comment:
No need to use apply here because the symbol and geometry can be specified as named arguments in the constructor
```suggestion
        Graphic(
            symbol = locationMarkerSymbol,
            geometry = startingPoint
        )
```
and same below

Reply:
You're right, perhaps I'm getting too caught up with Kotlin idioms 😬. I've refactored them to use constructors.

Reply:
I think that's pretty normal when working with a new language so I wouldn't worry about it 😊